oppijanumerorekisteri-domain/src/main/java/fi/vm/sade/oppijanumerorekisteri/repositories/impl/HenkiloRepositoryImpl.java

```diff
@@ -144,4 +148,60 @@
         });
         return henkiloDtoList;
     }
+
+    @Override
+    public List<HenkiloViiteDto> findHenkiloViitteesByHenkilo(HenkiloCriteria query) {
+        Where where = buildSqlConditions(query, "h");
+        SQLQuery q = em.unwrap(Session.class)
```

Comment:
Jos tämä menee vaikeakais, niin pitäisikö vain luopua koko aikarajauksesta ja laittaa API toimimaan niin, että sieltä saa kaikki viitteet haettua?

Se palvelisi ainoan käyttäjän valinta-tulos-henkiloviite-synchronizer :in tarpeita ihan hyvin, ja 11 sekuntia on siihen aivan mainiosti riittävä vasteaika, kun kyse on kerran yössä ajettavasta eräajosta.